EvolvSwiftSDK/EvolvSwiftSDKTests/Common-Tests/EvolvStoreNewTests.swift

```diff
@@ -235,3 +235,58 @@ class EvolvStoreNewTests: XCTestCase {
         XCTAssertEqual(firstActiveKeys, secondActiveKeys)
     }
 }
+
+// MARK: - Entry keys
+extension EvolvStoreNewTests {
+    func testEntryKeysAreEvaluated() {
+        let context = EvolvContextContainerImpl(remoteContextUserInfo: ["location" : "UA",
+                                                                        "view" : "home",
+                                                                        "signedin" : "yes"],
+                                                localContextUserInfo: [:])
+        
+        let evolvStore = initializeEvolvStore(with: context)
+        
+        let expectedActiveVariantKeys: Set = ["home:-3722956525781592630", "home.cta_text:921751162155200504"]
+        let actualActiveVariantKeys = evolvStore.activeVariantKeys.value
+        
+        XCTAssertEqual(expectedActiveVariantKeys, actualActiveVariantKeys)
+    }
+    
+    func testEntryKeysAreReevaluated() {
+        let context = EvolvContextContainerImpl(remoteContextUserInfo: ["location" : "UA",
+                                                                        "view" : "home",
+                                                                        "signedin" : "yes"],
+                                                localContextUserInfo: [:])
+        
+        let evolvStore = initializeEvolvStore(with: context)
+        
+        evolvStore.set(key: "view", value: "next", local: false)
+        
+        let expectedActiveVariantKeys: Set = ["next.layout:6424736096006099639", "next:-6123526860146466115"]
+        let actualActiveVariantKeys = evolvStore.activeVariantKeys.value
+        
+        XCTAssertEqual(expectedActiveVariantKeys, actualActiveVariantKeys)
+    }
+    
+    func testEntryKeysAreReevaluatedToEmptySet() {
+        let context = EvolvContextContainerImpl(remoteContextUserInfo: ["location" : "UA",
+                                                                        "view" : "home",
+                                                                        "signedin" : "yes"],
+                                                localContextUserInfo: [:])
+        
+        let evolvStore = initializeEvolvStore(with: context)
+        
+        evolvStore.set(key: "view", value: "none", local: false)
+        
+        let expectedActiveVariantKeys: Set<String> = []
+        let actualActiveVariantKeys = evolvStore.activeVariantKeys.value
+        
+        XCTAssertEqual(expectedActiveVariantKeys, actualActiveVariantKeys)
+    }
+    
+    func testHashCodeIsEvaluatedWithoutRuntimeOverflowError() {
+        // This line will throw an uncatchable runtime error
+        // if integer overflow is not allowed.
+        _ = "{\"cta_text\":\"Click Here\"}".evolvHashCode()
+    }
```

Comment:
In what circumstances would integer overflow not be allowed?

Reply:
[EvolvSwiftSDK/EvolvSwiftSDK/Sources/Extensions/Foundation+Extensions.swift, line 48](https://github.com/evolv-ai/ios-sdk/pull/20/files/800a43283191e53413a69bae4024a995d41cf405#diff-5a5c47074cabafb4def7844cc7718854b9a53ea054e0611c93d12d0e16004f35R48).
Integer overflow is only allowed with special operator such as "&*", "&+". By default, the app would crash if it encounters integer overflow.